modin/engines/base/io/text/csv_dispatcher.py

```diff
@@ -55,13 +58,32 @@ def _read(cls, filepath_or_buffer, **kwargs):
         if chunksize is not None:
             return cls.single_worker_read(filepath_or_buffer, **kwargs)
 
-        skiprows = kwargs.get("skiprows")
-        if skiprows is not None and not isinstance(skiprows, int):
+        # header can be skipped because header information is got before
+        # data file partitioning
+        header = kwargs.get("header", "infer")
+        names = kwargs.get("names", None)
+        header_size = cls._define_neader_size(header, names)
+
+        skiprows = kwargs.get("skiprows", None)
```

Comment:
Previously were used both `pop` and `get` methods (looks like it was code duplication). Now we should use `get` method because `skiprows` value should be kept in the `kwargs` dict for proper `names` parameter definition [here](https://github.com/modin-project/modin/blob/master/modin/engines/base/io/text/csv_dispatcher.py#L71-L74)